include/boost/geometry/views/detail/boundary_view/implementation.hpp

```diff
@@ -0,0 +1,400 @@
+        }
+        return range::at(interior_rings(*m_polygon), m_index - 1);
+    }
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    inline bool equal(polygon_rings_iterator
+                      <
+                          OtherPolygon,
+                          OtherValue,
+                          OtherReference,
+                          OtherDifference
+                      > const& other) const
+    {
+        return m_polygon == other.m_polygon && m_index == other.m_index;
+    }
+
+    inline void increment()
+    {
+        ++m_index;
+    }
+
+    inline void decrement()
+    {
+        --m_index;
+    }
+
+    template
+    <
+        typename OtherPolygon,
+        typename OtherValue,
+        typename OtherReference,
+        typename OtherDifference
+    >
+    inline Difference distance_to(polygon_rings_iterator
+                                  <
+                                      OtherPolygon,
+                                      OtherValue,
+                                      OtherReference,
+                                      OtherDifference
+                                  > const& other) const
+    {
+        return other.m_index - m_index;
+    }
+
+    inline void advance(Difference n)
+    {
+        m_index += n;
+    }
+
+private:
+    Polygon* m_polygon;
+    std::size_t m_index;
+};
+
+
+template <typename Ring>
+class ring_boundary : closeable_view<Ring, closure<Ring>::value>::type
+{
+private:
+    typedef typename closeable_view<Ring, closure<Ring>::value>::type base_type;
+
+public:
+    typedef typename base_type::iterator iterator;
+    typedef typename base_type::const_iterator const_iterator;
+    
+    typedef linestring_tag tag_type;
+
+    explicit ring_boundary(Ring& ring)
+        : base_type(ring) {}
+
+    iterator begin() { return base_type::begin(); }
+    iterator end() { return base_type::end(); }
+    const_iterator begin() const { return base_type::begin(); }
+    const_iterator end() const { return base_type::end(); }
+};
+
+
+template <typename Polygon>
+class polygon_boundary
+{
+    typedef boundary_view<typename ring_type<Polygon>::type> boundary_view_type;
+    typedef polygon_rings_iterator<Polygon> rings_iterator_type;
+
+    inline void initialize_views(Polygon& polygon)
+    {
+        std::pair<boundary_view_type*, std::ptrdiff_t> result
+            = std::get_temporary_buffer<boundary_view_type>(m_num_rings);
+
+        if ( result.second == 0 )
+        {
+            throw std::bad_alloc();
+        }
+
+        m_views = result.first;
+
+        std::ptrdiff_t ctr = 0;
+        rings_iterator_type first(polygon);
+        rings_iterator_type last(polygon, true);
+        for (rings_iterator_type rit = first; rit != last; ++rit, ++ctr)
+        {
+            new (m_views + ctr) boundary_view_type(*rit);
+        }
```

Comment:
This code is not exception-safe. If the constructor of boundary_view_type thrown at some point (because e.g. the creation of a dynamic array or vector failed inside) then the already created objects wouldn't be destroyed. You should rather use std::uninitialized_copy().